### DISCUSSION TITLE
Energy Usage Reporting

### DISCUSSION THREAD

**Discussion by klkohl** [PeteRager/lennoxs30#211] [Ideas]
It appears to me that the inverter energy is specific to the heat pump and does not include the air handler and auxillary heat unit. Is there a way to include those?

I've looked through the discussions and it appears that this was discussed a year ago, but wasn't resolved.

**Comment by PeteRager** [comment]
You are correct inverter power is the heat pump or air conditioner outdoor unit power.  It is derived from two equipment diagnostics ou_inverter_input_current and ou_inverter_input_voltage.  

There is a iu_indoor_blower_power diagnostic which is reported as a %.  To turn that into a power reading we’d need to know the maximum power of the blower (which is unit specific).  I haven’t been able to find this in the Lennox docs, they most talk about the max power requirements from a circuit breaker.   The best way may be to attach a current clamp to the supply line and take some measurements at different fan CFMs.  My understanding is it’s an exponential relationship   

As far as the aux heating unit is it supplying any diagnostic data that could be helpful?

**Reply by klkohl** [reply]
The only relevant diagnostic data is see is _iu_of_electric_heat_sections_on and _iu_heating_rate. It appears that there are 3 heat sections and that the heating rate is a percentage that corresponds to the number of heat sections. For 3 heat sections the heating rate is 100%. For 2 it's 66%. So it seems like we're in the same boat as the blower. I'm doing some research on my particular air handler to see if I can find some relevant information.

**Reply by klkohl** [reply]
I think I found the information we need:
https://resources.lennoxcommercial.com/resources/94c467f9-ac8e-405b-84e5-ff50ce771348-ehb_cba38mv_1808c.pdf

My blower is for the CBA38MV-048, see page 14. My auxiliary heat unit is ECB38-15CB and is shown on page 22.

**Comment by PeteRager** [comment]
On the blower info, I believe yours is fully variable like mine is.   And it is settable in 10 CFM increments from 200 CFM to 1100 CFM (approximate values), each additional 10 CFM uses more power.   I suppose it is says that the max it uses is 1025 watts, then a using a template you could multiple 1025 * % blower power and then use that?

For the electric furnace: a couple of questions.
- Does a device get properly created?  I’ve never seen one before. 
- I see you have the heat sections on the indoor unit, there’s not other data in the furnace device page?
- There is a aux heat attribute on the climate zones - does this turn on when the furnace is running?

Perhaps knowing the furnace kWh  (12,600 watts) and multiplying that by the heating rate % gives us the power?

On a related note, in the upcoming release I’m adding binary_sensors to indicate when the heat pump is locked out (due to being outside it’s rated ambient range)

**Reply by klkohl** [reply]
Yes, my blower is settable in 10 CFM increments, with a minimum of 250 and max of 2145 CFM.

Regarding the electric furnace (auxiliary heat):

- Does a device get properly created? I’ve never seen one before.

I'm not seeing a separate device, I'm only seeing a couple of entities in the air handler device.

- I see you have the heat sections on the indoor unit, there’s not other data in the furnace device page?

Reading the documentation, it appears that there are 3 "rows" of heating elements for the auxiliary heat. I'm presuming this corresponds to auxiliary heat sections and auxiliary heating rate in the diagnostics section for the air handler.

- There is a aux heat attribute on the climate zones - does this turn on when the furnace is running?

Yes.

The new sensor for ambient lockout should be helpful. I'm trying to get my head around when my system uses the auxiliary heat vs the heat pump. It's been in the mid 20's the last couple of mornings and it appears that the heat pump shuts down and I'm running on auxiliary heat only. It was my understanding when I bought my system this last summer that the heat pump would operate well below 32F. I did notice that when I turned off the heat this morning to test the aux heat attribute, that when I turned the system back on, only the heat pump came on.

Thanks for all your help and the work you've put into this.

**Reply by PeteRager** [reply]
Thanks for the notes, working on that lockout stuff today.

I believe there are equipment parameters that control the temps for heat pump lockout.  Its possible these were not set properly by the installer for your heat pump.  A history chart with the parameters, outdoor temp, hvac action and aux could be interesting.   I usually pull those attributes out into there own template sensors as it makes it easy to chart.

I’ve been looking at message log files and there is no information regarding the blower power rating or the furnace power rating.  So I don’t think the integration can provide those power readings unless we create a data set for each equipment model.  For now, what I’d propose is see if you can create the templates sensor for blower and furnace power based on the ratings in the doc and get that working.  Once you have that I’ll try it for my blower and also put a Current clamp on to see if the calculated values are close to actual.  I’m expecting the unit draws some amount of power all the time, I’d like to find out what that is.   Anyways, with that we can create an application note, provide sample template sensor and link in the docs you found?

**Reply by PeteRager** [reply]
Those new sensors for ambient lockout are available in latest release.   Give them a try when you get a chance, and let me know if they work correctly: as I don’t have a heat pump to directly test with.

**Reply by klkohl** [reply]
The ambient lockout sensors seem to work. Thanks for including those.

Regarding the active_alerts and alert sensors, I'm not clear on what those are showing. It appears that in my case the active_alerts are showing integers of 1, 2, or 3, with 1 on all the time. That is, it doesn't show 0. The alert sensor is just reporting "info".

The ou_Compressor_Active_Alarm is reporting, when it trips, is reporting either 14 or 29. In my case the 29 appears to correspond to an Alert Code 433.

Can you shed some light on this?

For what it's worth, I had my Lennox system installed this last June. Since it got cold about a month ago it's been not working great. So having this integration has been really helpful in trying to trouble shoot the problem. Thanks again for  all your work.

**Reply by PeteRager** [reply]
Good questions.  The ambient lockouts appear as alerts and will be in the alert list.   That may be why you are always seeing a 1?

The best way to see what is going on is to look at the attribute of the alarm list, that has the list of alerts from the controller.

Here is what mine looks like, I have the outdoor air conditioner breaker turned off for the winter, so have this  persistent alert.

![F8DD8AAB-3C79-4106-8D7F-6FE1C51ED986](https://user-images.githubusercontent.com/76050312/208732651-2459ef5a-eba9-4fbc-b06c-c8abf629718e.jpeg)

**Reply by PeteRager** [reply]
This Lennox docs describes the different error codes and what to do

https://tech.lennoxintl.com/C03e7o14l/VIu12Ch2uV/Corp1817-L8f.pdf
